pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -153,12 +154,30 @@ func MakeClusterSpec(nodeCount int, opts ...Option) ClusterSpec {
 
 // ClustersCompatible returns true if the clusters are compatible, i.e. the test
 // asking for s2 can reuse s1.
-func ClustersCompatible(s1, s2 ClusterSpec) bool {
-	s1.Lifetime = 0
-	s2.Lifetime = 0
+func ClustersCompatible(s1, s2 ClusterSpec, cloud string) bool {
+	// only consider the specification of the cloud that we are running in
+	clearClusterSpecFields(&s1, cloud)
+	clearClusterSpecFields(&s2, cloud)
 	return s1 == s2
 }
 
+func clearClusterSpecFields(cs *ClusterSpec, cloud string) {
+	cs.Lifetime = 0
+	structType := reflect.TypeOf(*cs)
+	for i := 0; i < structType.NumField(); i++ {
+		// Get field
+		field := structType.Field(i)
+		// Check if the "cloud" tag exists
+		if tag, ok := field.Tag.Lookup("cloud"); ok {
+			// Zero out struct if not GCE (for case cloud == GCE)
```

Comment:
This comment may not be applicable anymore, might be good to update it to the say the `cloud` parameter. `zero out struct if not target cloud`